web/services/createTemplatesService.php

```diff
@@ -171,55 +175,29 @@
-                DBPostgres::boolToString($template["onsite"]) . ", " .
-                DBPostgres::checkStringNull($template["text"]) . ", " .
-                DBPostgres::checkStringNull($template["ttype"]) . ", " .
-                DBPostgres::checkNull($template["userId"]) . ", " .
-                DBPostgres::checkNull($template["projectId"]) . ", " .
-                DBPostgres::checkNull($template["customerId"]). ", " .
-                DBPostgres::checkNull($template["taskStoryId"]) .")";
-
-            $res = pg_query($connect, $sql);
-            if ($res == NULL)
-                $string = "<return service='createTemplates'><success>false</success><error id='1'>There was some error while creating the tasks</error></return>";
-
-            $createTemplates[$key]["id"] = DBPostgres::getId($connect, "template_id_seq");
-        }
+        if (TemplatesFacade::CreateTemplates($createTemplates) == -1)
+            $string = "<return service='createTemplates'><success>false</success><error id='1'>There was some error while creating the tasks</error></return>";
+
         if (!$string)
         {
             $string = "<return service='createTemplates'><success>true</success><ok>Operation Success!</ok><templates>";
             foreach ($createTemplates as $template) {
-                $string .= "<template><id>{$template['id']}</id>";
-                $string .= "<name>{$template["name"]}</name>";
-                $string .= "<story>{$template["story"]}</story>";
-                $string .= "<telework>{$template["telework"]}</telework>";
-                $string .= "<onsite>{$template["onsite"]}</onsite>";
-                $string .= "<text>{$template["text"]}</text>";
-                $string .= "<ttype>{$template["ttype"]}</ttype>";
-                $string .= "<userId>{$template["userId"]}</userId>";
-                $string .= "<projectId>{$template["projectId"]}</projectId>";
-                $string .= "<customerId>{$template["customerId"]}</customerId>";
-                $string .= "<taskStoryId>{$template["taskStoryId"]}</taskStoryId></template>";
+                $string .= "<template><id>{$template->getId()}</id>";
+                $string .= "<name>{$template->getName()}</name>";
+                $string .= "<story>{$template->getStory()}</story>";
+                $string .= "<telework>{$template->isTelework()}</telework>";
+                $string .= "<onsite>{$template->isOnsite()}</onsite>";
+                $string .= "<text>{$template->getText()}</text>";
+                $string .= "<ttype>{$template->getTtype()}</ttype>";
+                $string .= "<userId>{$template->getUserId()}</userId>";
+                $string .= "<projectId>{$template->getProjectId()}</projectId>";
+                $string .= "<customerId>{$template->getCustomerId()}</customerId>";
+                $string .= "<taskStoryId>{$template->getTaskStoryId()}</taskStoryId></template>";
```

Comment:
I've just had an idea. Could you create a method TemplateVO->toXml() which would do this? You could also reuse it in `getUserTemplatesService.php`. We could extend it to existing VOs in the future.